src/scripts/carousel.js

```diff
@@ -0,0 +1,129 @@
+(function() {
+  "use strict";
+
+  // Variables
+
+  var pBox = document.querySelectorAll(".carousel-box");
+  var pCarousel = document.querySelector(".products-carousel");
+  var carouselDots = document.querySelector(".carousel-dots");
+  var dotsMain = document.querySelector(".dots-main");
+  var boxArray = [];
+
+  // Dynamically added dots that depend on the number of product boxes
+
+  function addDots() {
+    var box;
+    var vPortWidth = window.screen.availWidth;
+    var pBoxL = pBox.length;
+    var i;
+    var dotsTempl = '<li><span class="dot-click basic-animation"></span></li>';
+    carouselDots.innerHTML = "";
+
+    if (vPortWidth >= 1025) {
+      box = pBoxL / 4;
+      for (i = 0; i < box; i++) {
+        carouselDots.insertAdjacentHTML("beforeend", dotsTempl);
+      }
+      return boxArray.push(box);
+    } else if (vPortWidth >= 990) {
+      box = pBoxL / 3;
+      for (i = 0; i < box; i++) {
+        carouselDots.insertAdjacentHTML("beforeend", dotsTempl);
+      }
+      return boxArray.push(box);
+    } else if (vPortWidth >= 568) {
+      box = pBoxL / 2;
+      for (i = 0; i < box; i++) {
+        carouselDots.insertAdjacentHTML("beforeend", dotsTempl);
+      }
+      return boxArray.push(box);
+    } else if (vPortWidth <= 567) {
+      box = pBoxL;
+      for (i = 0; i < box; i++) {
+        carouselDots.insertAdjacentHTML("beforeend", dotsTempl);
+      }
+      return boxArray.push(box);
+    }
+  }
+
+  addDots();
+
+  // Resolution listener
+
+  window.onresize = addDots;
+
+  // variables whose state must be update
+
+  var dotClic = document.querySelectorAll(".dot-click");
+  var dotClick = dotClic.length;
+  var onActive = document.querySelectorAll(".panel-bar .dots li span");
+  var pBoxWidth = document.querySelectorAll(".carousel-box");
+  var move = pBoxWidth[0].clientWidth;
+  var viewWidth = window.screen.availWidth;
+  onActive[0].classList.add("active");
+
+  // This function adjusts the carousel movement
+
+  function setCount(a, b, c, d) {
+    var slide = (pCarousel.style.transform =
+      "translateX(-" + a * (d / c) + "px)");
+    if (b >= 1025) {
```

Comment:
nie rozumiem tego warunku bo zawsze zwracasz slide, więc po co nam te ifki?

Reply:
To wszystko przez zmienną 'b' która jest zależna od rozdzielczości i musi przeliczyć wszystko jeszcze raz (za każdym razem wynik jest inny)

Reply:
ale zobacz że obojętnie jaka będzie wartość b to i tak zawsze zwracasz slide który jest niezależny od b